File: csharp-beginner-workbook/module-06-oop-intro/examples/example-02-constructor.cs
// Example 2: Constructor
// This demonstrates creating a class with a constructor that initializes objects

// Define the Product class with constructor
class Product
{
    public string Name { get; set; }
    public double Price { get; set; }
    
    // Constructor - requires name and price when creating a Product
    public Product(string name, double price)
    {
        Name = name;
        Price = price;
    }
}

// Main program - create products using the constructor
Product apple = new Product("Red Apple", 0.99);
Product bread = new Product("Whole Wheat Bread", 2.50);
Product milk = new Product("Whole Milk", 3.99);

Console.WriteLine("=== Product Catalog ===");
Console.WriteLine($"{apple.Name}: ${apple.Price:F2}");
Console.WriteLine($"{bread.Name}: ${bread.Price:F2}");
Console.WriteLine($"{milk.Name}: ${milk.Price:F2}");

Console.WriteLine("\n=== Product Details ===");
Console.WriteLine($"Product 1 - Name: {apple.Name}, Price: ${apple.Price:F2}");
Console.WriteLine($"Product 2 - Name: {bread.Name}, Price: ${bread.Price:F2}");
Console.WriteLine($"Product 3 - Name: {milk.Name}, Price: ${milk.Price:F2}");

File: python-beginner-workbook/module-06-oop/examples/example-02-constructor.py
# example-02-constructor.py
# Demonstrates constructor with parameters

class Product:
    def __init__(self, name, price):
        self.name = name
        self.price = price
    
    def display_info(self):
        print(f"{self.name}: £{self.price:.2f}")

# Create products
laptop = Product("Laptop", 999.99)
mouse = Product("Mouse", 29.99)

laptop.display_info()
mouse.display_info()

